### ISSUE TITLE
Remarks about the API

### ISSUE THREAD

**Issue by sork** [jeremyckahn/rekapi#9]
Here are a few thoughts and suggestions:
1. Because Rekapi was created at first with HTML5 canvas in mind, with other rendering methods added at a later time, I feel that the API is currently a bit too canvas-oriented. All methods beginning with `canvas_` in the Kapi object seem to be out of place. I think that these should be relocated to a separate canvas-specific class.
   
   This line about `canvas_clear` in the documentation strikes me as a good example of this issue:
   _Erase the canvas. This only does something if Kapi is bound to an HTML 5 canvas_
   This means that this method makes no sense in other contexts. 
2. This leads to me suggest that there should be a canvas extension, just like there is a DOM extension. In my opinion, Rekapi should not assume that canvas is the preferred rendering method since - as you explained in your talk - Rekapi is about animating via keyframes, not rendering.
3. Therefore, some code should moved from `core` into this new canvas extension. For instance, I don't think the core Rekapi module should include the current `draw` function logic. When you think about it, the core is about animating actors, not about drawing. In particular, the drawOrder code looks useless to me if we are using Rekapi for DOM animations or even WebGL/SVG (future extensions!).

To sum it up, I think the Rekapi API and code architecture should follow a more generic approach and not make any assumptions about how the user will want to draw the animation. I'm certain that I've overlooked some things since I've only had a quick look at the code and the API, but I'd be interested in reading your opinion on this.

Related idea: Add a build script accepting different parameters allowing to create a version of Rekapi with your preferred rendering method, i.e. choose which extension(s) you want to use.
